### PR TITLE
Support of USB bind identification for iseq power supplies

### PR DESCRIPTION
**Pull request by mjmucha**
The iseq SHQ 122M does not support the standard `*IDN?` query. Therefore this PR adds the functionality to query/read the identifier of the SHQ 122M via the `#` command. Additionally included is a small fix for an bug with memorized_binds.

### CODE REVIEW

File: basil/utils/USBBinds.py [hunk 1]
```diff
@@ -31,12 +31,16 @@ def query_identification(rm, resource, baud_rate, read_termination=None, write_t
...
         # This retries the query a second time, since some devices do not answer the first time.
         # If a second exception arrises, it will be handled in calling function.
-        reply = inst.query("*IDN?", delay=0.1)
-    return reply
+        reply = inst.query(indentifer_cmd, delay=0.1)
+        
+    if read_after_query:
+        return inst.read()
+    else:
+        return reply
```
Comment (cbespin) [inline basil/utils/USBBinds.py:43]: Why is this needed?

File: basil/utils/USBBinds.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 import ruamel.yaml
 
 
-def query_identification(rm, resource, baud_rate, read_termination=None, write_termination=None, timeout=1000 * 5):
+def query_identification(rm, resource, baud_rate, read_termination=None, write_termination=None, timeout=1000 * 5, indentifer_cmd="*IDN?", read_after_query=False):
```
Comment (cbespin) [inline basil/utils/USBBinds.py:12]: Typo in `indentifier`

File: basil/utils/USBBinds.py [hunk 3]
```diff
@@ -105,7 +109,10 @@ def find_usb_binds(rm, log,
                     log.debug(f"Found memorized bind {res}")
                     result = memorized_binds[res]
                 else:
-                    result = query_identification(rm, res, instrument['baud_rate'], instrument['read_termination'], instrument['write_termination'], timeout=timeout)
+                    if instrument["type"].lower() == "iseg_hv":
```
Comment (cbespin) [inline basil/utils/USBBinds.py:112]: Maybe make a global "devices_with_non_default_values" dict, so that it is easier to potentially add more devices there instead of blowing up an `if` clause?

File: basil/utils/USBBinds.py [hunk 4]
```diff
@@ -105,7 +109,10 @@ def find_usb_binds(rm, log,
...
+                        result = query_identification(rm, res, instrument['baud_rate'], instrument['read_termination'], instrument['write_termination'], timeout=timeout, indentifer_cmd="#", read_after_query=True)
```
Comment (cbespin) [inline basil/utils/USBBinds.py:113]: Same typo as above

File: basil/utils/USBBinds.py [hunk 5]
```diff
@@ -228,6 +235,8 @@ def modify_basil_config(conf, log, skip_binds=[], save_modified=None):
             port = None
 
         instruments.append({
+            "name": tf["name"],
+            "type": "", # Standard instruments with *IQN? command
```
Comment (cbespin) [inline basil/utils/USBBinds.py]: What is happening here exactly? Is this a way to define different devices with non-`*IDN?` identifier commands?

**Review by cbespin — CHANGES_REQUESTED** [review]

File: basil/utils/USBBinds.py (hunk 1 above)
Comment (mjmucha) [reply basil/utils/USBBinds.py:43]: Because the SHQ 122m returns as query response the command string as check if the command was executed successfully. To get the return value of an command (or in this case the identification string) one has to additionally perform a read operation.

File: basil/utils/USBBinds.py (hunk 5 above)
Comment (mjmucha) [reply basil/utils/USBBinds.py]: This is done to identify serial transfer layers which uses implemented non-`*IDN?` devices. Without the addition of "name" and "type" in this temporary instrument dictionary the linking in the code block below would not work and thus also not the logic which you commented in your third comment.

**TL;DR:** The serial transfer layer does not know if it is used for e.g. an SHQ 122M. Therefore this part is needed to identify an instrument correctly.
